src/context/toastNotificationProvider.tsx

```diff
@@ -0,0 +1,209 @@
+  failure: (
+    title: string,
+    error: unknown,
+    message?: ReactNode,
+    actions?: NotificationAction[],
+  ) => ToastNotificationType;
+  clear: (notification?: ToastNotificationType[]) => void;
+  toggleListView: () => void;
+  isListView: boolean;
+  countBySeverity: GroupedNotificationCount;
+};
+
+export type GroupedNotificationCount = {
+  [key in ValueOf<typeof NotificationSeverity>]?: number;
+};
+
+const initialNotification: ToastNotificationType = {
+  id: "",
+  message: "",
+  type: "positive",
+};
+const ToastNotificationContext = createContext<ToastNotificationHelper>({
+  notifications: [],
+  success: () => initialNotification,
+  info: () => initialNotification,
+  failure: () => initialNotification,
+  clear: () => null,
+  toggleListView: () => null,
+  isListView: false,
+  countBySeverity: {},
+});
+
+interface Props {}
+
+const ToastNotificationProvider: FC<PropsWithChildren<Props>> = ({
+  children,
+}) => {
+  const [notifications, setNotifications] = useState<ToastNotificationType[]>(
+    [],
+  );
+  const [showList, setShowList] = useState(false);
+  const [notificationTimer, setNotificationTimer] =
+    useState<NodeJS.Timeout | null>(null);
+
+  // cleanup on timer if unmounted, not likely though since this component sits above App
+  useEffect(() => {
+    return () => {
+      if (notificationTimer) {
+        clearTimeout(notificationTimer);
+      }
+    };
+  }, []);
+
+  const showNotificationWithDelay = () => {
+    setNotificationTimer((prevTimer) => {
+      if (prevTimer) {
+        clearTimeout(prevTimer);
+      }
+
+      if (!showList) {
+        return setTimeout(() => {
+          setNotificationTimer(null);
+        }, HIDE_NOTIFICATION_DELAY);
+      }
+
+      return null;
+    });
+  };
+
+  const clearNotificationTimer = () => {
+    setNotificationTimer((prevTimer) => {
+      if (prevTimer) {
+        clearTimeout(prevTimer);
+      }
+      return null;
+    });
+  };
+
+  const addNotification = (
+    notification: NotificationType & { error?: unknown },
+  ) => {
+    const notificationToAdd = {
+      ...notification,
+      timestamp: new Date().toLocaleString(),
+      id: Date.now().toString() + (Math.random() + 1).toString(36).substring(7),
+    };
+
+    setNotifications((prev) => {
+      return [...prev, notificationToAdd];
+    });
+
+    showNotificationWithDelay();
+
+    return notificationToAdd;
+  };
+
+  const clear = (notifications?: ToastNotificationType[]) => {
+    if (!notifications) {
+      setNotifications([]);
+      setShowList(false);
+      clearNotificationTimer();
+      return;
+    }
+
+    const removeIdLookup: { [id: string]: boolean } = {};
+    notifications.forEach((notification) => {
+      removeIdLookup[notification.id] = true;
+    });
+
+    setNotifications((prev) => {
+      const newNotifications = prev.filter(
+        (existingNotification) => !removeIdLookup[existingNotification.id],
+      );
```

Comment:
Probably simpler to use a set?
```suggestion
    setNotifications((prev) => {
      const removeIdLookup = new Set(notifications);
      const newNotifications = prev.filter((item) => removeIdLookup.has(item));
```

Reply:
nice! :+1: